File: L003/digitsOfANo.java
import java.util.Scanner;

public class digitsOfANo {

    public static Scanner sc = new Scanner(System.in);

    public static void digitsOfANo_(int no) {
        int nod = countOfDigits(no);

        int div = (int)Math.pow(10 , nod-1);

        while(div != 0 ){
            int quot = no / div;
            System.out.println(quot);
            no %= div;
            div /= 10;
        }

    }

    public static int countOfDigits(int no) {

        int count = 0;
        while(no != 0){
            no = no/10;
            count++;
        }
        return count;
    }

    public static void main(String[] args) {

        int n = sc.nextInt();

        int no = n;

        digitsOfANo_(no);
    }
}

File: L003/digitsOfANo.cpp
#include<iostream>
#include<math.h>
using namespace std;

      
int countOfDigits(int num) {

        int count = 0;
        while(num != 0){
            num  = num/10;
            count++;
        }
        return count;
    }

void digitsOfANo_(int no) {

        int nod = countOfDigits(no);

        int div = pow(10 , nod-1);

        while(div != 0 ){
            int quot = no / div;
            cout<<quot<<endl;
            no %= div;
            div /= 10;
        }

    }
     int main() {

        int n;
        cin>>n;

        int no = n;

        digitsOfANo_(no);

        return 0;
    }